samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentEntityProducer.java

```diff
@@ -776,34 +811,42 @@ private List<String> getAttachmentResourceIds(NodeList list) {
 	
 	private <T> boolean migrateText(AssessmentService assessmentService, String toContext, T item, String itemHash,
 			boolean hasCaches,boolean hasDuplicates, boolean copyAttachments, String cacheCode, Map<String, String> textCache,
-			Set<Entry<String, String>> entrySet, Function<T, String> getter, BiConsumer<T, String> setter) {
+			Set<Entry<String, String>> entrySet, Map<String, String> transversalMap, MergeConfig mcx,
+			Function<T, String> getter, BiConsumer<T, String> setter) {
 
+		log.debug("migrateText: {} {}", itemHash, copyAttachments);
 		String cacheKey = itemHash + "-" + cacheCode;
 
 		if (hasCaches && textCache.containsKey(cacheKey)) {
-			// Item instruction has been cashed, lets get it form the cache
+			// Item instruction has been cached, lets get it from the cache
 			setter.accept(item, textCache.get(cacheKey));
 			return true;
 		} else {
 			// Item instruction has not been cached, lets try migrating
 			String itemText = StringUtils.trimToEmpty(getter.apply(item));
 			String migratedText;
-			if (copyAttachments) {
-				migratedText = assessmentService.copyContentHostingAttachments(itemText, toContext);
+			log.debug("itemText before {}", itemText);
+			if ( mcx != null ) {
+				migratedText = ltiService.fixLtiLaunchUrls(itemText, toContext, mcx);
+				migratedText = linkMigrationHelper.migrateLinksInMergedRTE(toContext, mcx, migratedText);
 			} else {
-				migratedText = itemText;
+				if (copyAttachments) {
+					migratedText = assessmentService.copyContentHostingAttachments(itemText, toContext);
+				} else {
+					migratedText = itemText;
+				}
+				migratedText = linkMigrationHelper.migrateAllLinks(entrySet, migratedText);
+				String fromContext = null;
```

Comment:
Is that correct? If yes, we might just pass null, I think that would be less confusing.

Reply:
I like that pattern because it makes it really clear that sending null is a 100% OK thing.   I really dislike calls with `null, null, null` in their calling sequence.   We lose why each of those is OK to be null and what each of the null s really is.  I know that in modern IDEs - it tells you what each of the nulls represents.  So I will change it :)